ddl/index.go

```diff
@@ -940,7 +942,60 @@ func (w *worker) addTableIndex(t table.Table, indexInfo *model.IndexInfo, reorgI
 	}
 	defer closeAddIndexWorkers(idxWorkers)
 
-	return w.buildKVRangesIndex(t, idxWorkers, job, reorgInfo)
+	finish := false
+	for !finish {
+		err := w.buildKVRangesIndex(t, idxWorkers, job, reorgInfo)
```

Comment:
Rename `buildKVRangesIndex` to `buildKVRangesIndexByPartition` or add some comment for it?